ang/civicase/case/directives/case-overview.directive.js

```diff
@@ -72,58 +71,111 @@
+     * @param {object} caseFilters parameters to use for filtering the stats data.
      */
-    $scope.toggleStatusVisibility = function ($event, index) {
-      $scope.caseStatuses[index + 1].isHidden = !$scope.caseStatuses[index + 1].isHidden;
-      storeHiddenCaseStatuses();
-      $event.stopPropagation();
-    };
+    function caseFilterWatcher (caseFilters) {
+      loadStatsData(caseFilters);
+      loadCaseTypes(caseFilters)
+        .then(function () {
+          var caseStatusNames = getCaseStatusNamesBelongingToCaseTypes($scope.caseTypes);
+
+          $scope.showBreakdown = $scope.caseTypes.length <=
+            MAXIMUM_CASE_TYPES_TO_DISPLAY_BREAKDOWN;
+          $scope.caseStatuses = getSortedCaseStatusesByName(caseStatusNames);
+
+          $scope.$emit('civicase::custom-scrollbar::recalculate');
+        });
+    }
 
     /**
-     * Toggles the visibility of the breakdown dropdown
+     * Given a list of case types, it will return a unique list of
+     * case status names as defined for each one of the case types.
+     *
+     * Note: When a case type supports all statuses, it does not store any status
+     * names under `definition.statuses`. If the statuses definition is empty we
+     * must assume the case supports all statuses.
+     *
+     * @param {object[]} caseTypes a list of case type objects.
+     * @returns {string[]} a list of case sttus names.
      */
-    $scope.toggleBrekdownVisibility = function () {
-      $scope.showBreakdown = !$scope.showBreakdown;
-    };
+    function getCaseStatusNamesBelongingToCaseTypes (caseTypes) {
+      return _.chain(caseTypes)
+        .map(function (caseType) {
+          return caseType.definition.statuses
+            ? caseType.definition.statuses
+            : allCaseStatusNames;
+        })
+        .flatten()
+        .unique()
+        .value();
+    }
 
     /**
-     * Watcher function for caseFilter
+     * @param {string[]} caseStatusNames a list of case status names.
+     * @returns {object[]} the full case status details belonging to the
+     *   given case status names.
      */
-    function caseFilterWatcher () {
-      getCaseTypes();
-      loadStatsData();
+    function getSortedCaseStatusesByName (caseStatusNames) {
+      return _.chain(caseStatusNames)
+        .map(function (caseStatusName) {
+          return caseStatusesIndexedByName[caseStatusName];
+        })
+        .sortBy('weight')
+        .indexBy('value')
+        .value();
+    }
+
+    /**
+     * Get Case Types based on filters
+     *
+     * @param {object} caseFilters parameters to use for filtering case types.
+     * @returns {Promise} promise
+     */
+    function loadCaseTypes (caseFilters) {
+      var params = {
+        sequential: 1,
+        case_type_category: caseFilters['case_type_id.case_type_category'],
+        id: caseFilters.case_type_id,
+        is_active: 1
+      };
+
+      return crmApi('CaseType', 'get', params)
```

Comment:
`CaseType` is already available from `CRM['civicase-base']`, cant we use that and then filter? instead of calling the api.